src/lib/mdns/minimal/records/Txt.h

```diff
@@ -0,0 +1,68 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <string.h>
+
+#include <mdns/minimal/records/ResourceRecord.h>
+
+namespace mdns {
+namespace Minimal {
+
+class TxtResourceRecord : public ResourceRecord
+{
+public:
+    static constexpr uint64_t kDefaultTtl = 10;
+
+    TxtResourceRecord(const FullQName & qName, const char ** entries, size_t entryCount) :
+        ResourceRecord(QType::TXT, qName), mEntries(entries), mEntryCount(entryCount)
+    {
+        SetTtl(kDefaultTtl);
+    }
+
+    template <size_t N>
+    TxtResourceRecord(const FullQName & qName, const char * (&entries)[N]) :
+        ResourceRecord(QType::TXT, qName), mEntries(entries), mEntryCount(N)
+    {
+        SetTtl(kDefaultTtl);
+    }
+
+protected:
+    bool WriteData(chip::BufBound & out) const override
+    {
+        for (size_t i = 0; i < mEntryCount; i++)
+        {
+            size_t len = strlen(mEntries[i]);
+            if (len > kMaxQNamePartLength)
+            {
+                return false;
+            }
+
+            out.Put8(static_cast<uint8_t>(len));
+            out.Put(mEntries[i]);
+        }
+        return out.Fit();
+    }
+
+private:
+    const char ** mEntries;
```

Comment:
I'd suggest to use a struct rather than `char*` for text entry since the mDNS RFC allows the val to contain `\0` in the `val` part of the `key=val`. 
I'd also suggest to copy the entry here so that the user won't worry about the lifecycle of the memory.

Reply:
I would prefer to iterate over these. Reasons being:
  - None of the CHIP spec will have a \0 inside the key or value, so we probably do not care about those
  - Copying requires memory management. Current implementation requires no dynamic memory allocation at all. If allowed, we would be able to simplify things a lot (e.g. templated 'add record/responder') however I avoided that on purpose.